packages/calcite-components/src/components/block/block.tsx

```diff
@@ -387,13 +398,26 @@ export class Block extends LitElement implements InteractiveComponent, OpenClose
     this.hasContentStart = slotChangeHasAssignedElement(event);
   }
 
+  private updateBlockSectionChildren(event: Event): void {
+    this.blockSectionChildren = (event.target as HTMLSlotElement)
```

Comment:
I think you could simplify this by using the [`slotChangeGetAssignedElements`](https://github.com/Esri/calcite-design-system/blob/fcf0b0c61bda6d86da4c392582d4930a99f63533/packages/calcite-components/src/utils/dom.ts#L529) DOM util.